app/window.py

```diff
@@ -873,21 +873,10 @@ def refresh(self):
   def setTextBuffer(self, textBuffer):
     app.log.info('setTextBuffer')
     #self.normalize()
-    if self.textBuffer is not None:
-      app.history.set(['files', self.textBuffer.fullPath, 'cursor'],
-          (self.textBuffer.penRow, self.textBuffer.penCol))
-    # TODO(dschuyler): Do we need to restore positions and selections here?
     self.controller.setTextBuffer(textBuffer)
     Window.setTextBuffer(self, textBuffer)
+    self.textBuffer.restoreUserHistory()
```

Comment:
Yeah it was primarily because there was no view that I moved it here. I'll see if I can scroll to the cursor, but I think its better if we center the cursor maybe?